redskyctl/internal/commands/run/view.go

```diff
@@ -0,0 +1,420 @@
+/*
+Copyright 2021 GramLabs, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package run
+
+import (
+	"errors"
+	"fmt"
+	"os/exec"
+	"text/tabwriter"
+
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/thestormforge/optimize-controller/redskyctl/internal/commands/run/form"
+	"github.com/thestormforge/optimize-controller/redskyctl/internal/commands/run/internal"
+	"github.com/thestormforge/optimize-controller/redskyctl/internal/commands/run/out"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+const (
+	ScenarioTypeStormForger = "StormForge"
+	ScenarioTypeLocust      = "Locust"
+)
+
+var pagerKeyBindings = []out.KeyBinding{
+	{
+		Key:  tea.Key{Type: tea.KeyCtrlX},
+		Desc: "Quit",
+	},
+	{
+		Key:  tea.Key{Type: tea.KeySpace},
+		Desc: "Next page",
+	},
+	{
+		Key:  tea.Key{Type: tea.KeyRunes, Runes: []rune{'b'}},
+		Desc: "Previous page",
+	},
+}
+
+// initializeModel is invoked before the program is started to ensure things
+// are in a valid state prior to starting.
+func (o *Options) initializeModel() {
+	var opts []out.FieldOption
+	if o.Verbose {
+		opts = append(opts, out.VerbosePrompts)
+	}
+
+	o.generatorModel.ScenarioType = out.FormField{
+		Prompt:       "Where do you want to get your load test from?",
+		Instructions: []string{"up/down: select", "pgup: back", "enter: continue"},
+		Choices: []string{
+			ScenarioTypeStormForger,
+			ScenarioTypeLocust,
+		},
+	}.NewChoiceField(opts...)
+	o.generatorModel.ScenarioType.Select(0)
+
+	o.generatorModel.StormForgerTestCaseInput = out.FormField{
+		Prompt:         "Please select a load test to optimize:",
+		LoadingMessage: "Fetching test cases from StormForger",
+		Instructions:   []string{"up/down: select", "pgup: back", "enter: continue"},
+	}.NewChoiceField(opts...)
+
+	o.generatorModel.StormForgerGettingStarted = out.FormField{
+		Prompt: `Check this out to see how you can get set up with a StormForge load test:
+https://docs.stormforger.com/guides/getting-started/`,
+	}.NewExitField(opts...)
+
+	o.generatorModel.LocustfileInput = out.FormField{
+		Prompt:          "Please input a path to your Locust load test to optimize:",
+		Placeholder:     "( e.g. ~/my-project/tests/locustfile.py )",
```

Comment:
does this need to be an absolute path or will relative work?

Reply:
Either or. We set the application path in `ReadApplication` in `run.go` to the wd so it will resolve against there.